frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -27,3 +42,89 @@ frappe.ui.form.on("System Settings", "enable_two_factor_auth", function(frm) {
+    if ((frm.doc.enable_backup_encryption && frm.doc.new_password_encryption != frm.doc.confirm_password_encryption) || !frm.doc.confirm_password_encryption) {
+
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+
+    } else {
+        frm.doc.new_password_encryption = "";
+        frm.doc.confirm_password_encryption = "";
+        frm.doc.password_encryption = "";
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
+
+        } else {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: red"
+
+        }
+
+    }
+
+});
+
+frappe.ui.form.on("System Settings", "change_encryption_password", function (frm) {
+    if (frm.doc.enable_backup_encryption && frm.doc.change_encryption_password) {
+        frm.fields_dict.new_password_encryption.df.reqd = 1;
+        frm.fields_dict.new_password_encryption.df.hidden = 0;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 1;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 0;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.new_password_encryption.input_area.children[0].value = "******";
+        cur_frm.fields_dict.confirm_password_encryption.input_area.children[0].value = "******";
+
+
+    } else {
+        frm.fields_dict.new_password_encryption.df.reqd = 0;
+        frm.fields_dict.new_password_encryption.df.hidden = 1;
+        frm.refresh_field("new_password_encryption");
+        frm.fields_dict.confirm_password_encryption.df.reqd = 0;
+        frm.fields_dict.confirm_password_encryption.df.hidden = 1;
+        frm.refresh_field("confirm_password_encryption");
+        cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = null;
+    }
+    cur_frm.fields_dict.confirm_password_encryption.$input[0].onchange = function () {
+
+        if (frm.doc.new_password_encryption == frm.doc.confirm_password_encryption) {
+            $('input[data-fieldname="confirm_password_encryption"]')[0].style = "border-color: #08c708"
```

Comment:
Is it necessary to have this here in the code? What is it accomplishing? Can you please add some comments where the intent is not really obvious.

Reply:
we've mode this js function into the ```if condition``` that is located above where it is initialized before, so it will only be initialized if the option for encryption and changing the password is selected. This is not really necessary, but this will indicate if the two password fields are matched. if not, a red border of the confirm password field will be shown otherwise, green. If you think this is really not necessary, we can remove this. or maybe you can suggest a better way to handle this purpose, that would be very much appreciated.

Reply:
the method is now removed for every condition. We're now using frappe.ui.form.on method to handle this process and initialized outside, thanks for the suggestion below.